File: wsTripXML/Code/OTAClasses/OTACruiseCategoryAvail.vb
Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization

Namespace wsTravelTalk.wmCruiseCategoryAvail

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Budget

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PricingType As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MinPrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MaxPrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public GuidelinePrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CurrencyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DecimalPlaces As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Currency

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CurrencyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DecimalPlaces As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class InclusivePackageOption

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CruisePackageCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public InclusiveIndicator As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public StartDate As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedSailing

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VoyageID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Start As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Duration As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public [End] As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="NMTOKEN")> _
        Public Status As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SailingInfo

        '<remarks/>
        Public SelectedSailing As SelectedSailing

        '<remarks/>
        Public InclusivePackageOption As InclusivePackageOption

        '<remarks/>
        Public Currency As Currency
    End Class

End Namespace

File: src/wsTripXML/Code/OTAClasses/OTACruiseCategoryAvail.cs

using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk.wmCruiseCategoryAvail
{

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Budget
    {

        // <remarks/>
        [XmlAttribute()]
        public string PricingType;

        // <remarks/>
        [XmlAttribute()]
        public string MinPrice;

        // <remarks/>
        [XmlAttribute()]
        public string MaxPrice;

        // <remarks/>
        [XmlAttribute()]
        public string GuidelinePrice;

        // <remarks/>
        [XmlAttribute()]
        public string CurrencyCode;

        // <remarks/>
        [XmlAttribute()]
        public string DecimalPlaces;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Currency
    {

        // <remarks/>
        [XmlAttribute()]
        public string CurrencyCode;

        // <remarks/>
        [XmlAttribute()]
        public string DecimalPlaces;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class InclusivePackageOption
    {

        // <remarks/>
        [XmlAttribute()]
        public string CruisePackageCode;

        // <remarks/>
        [XmlAttribute()]
        public string InclusiveIndicator;

        // <remarks/>
        [XmlAttribute()]
        public string StartDate;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedSailing
    {

        // <remarks/>
        [XmlAttribute()]
        public string VoyageID;

        // <remarks/>
        [XmlAttribute()]
        public string Start;

        // <remarks/>
        [XmlAttribute()]
        public string Duration;

        // <remarks/>
        [XmlAttribute()]
        public string End;

        // <remarks/>
        [XmlAttribute()]
        public string VendorCode;

        // <remarks/>
        [XmlAttribute()]
        public string VendorName;

        // <remarks/>
        [XmlAttribute()]
        public string ShipCode;

        // <remarks/>
        [XmlAttribute()]
        public string ShipName;

        // <remarks/>
        [XmlAttribute(DataType = "NMTOKEN")]
        public string Status;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SailingInfo
    {

        // <remarks/>
        public SelectedSailing SelectedSailing;

        // <remarks/>
        public InclusivePackageOption InclusivePackageOption;

        // <remarks/>
        public Currency Currency;
    }

}